addons/web/static/tests/mock_server_tests.js

```diff
@@ -94,3 +112,102 @@ QUnit.test("performRPC: search_read with active_test=true", async function (asse
     });
     assert.deepEqual(result, [{ id: 1, name: "Jean-Michel" }]);
 });
+
+QUnit.test("performRPC: read_group, group by date", async function (assert) {
+    assert.expect(10);
+    const server = new MockServer(this.data, {});
+    let result = await server.performRPC("", {
+        model: "bar",
+        method: "read_group",
+        args: [[]],
+        kwargs: {
+            fields: ["foo"],
+            domain: [],
+            groupby: ["date"], //Month by default
+        },
+    });
+    assert.deepEqual(
+        result.map((x) => x.date),
+        ["December 2016", "October 2016", "April 2016"]
+    );
+    assert.deepEqual(
+        result.map((x) => x.date_count),
+        [3, 1, 1]
+    );
+
+    result = await server.performRPC("", {
+        model: "bar",
+        method: "read_group",
+        args: [[]],
+        kwargs: {
+            fields: ["foo"],
+            domain: [],
+            groupby: ["date:day"],
+        },
+    });
+    assert.deepEqual(
+        result.map((x) => x["date:day"]),
+        ["2016-12-14", "2016-10-26", "2016-12-15", "2016-04-11"]
+    );
+    assert.deepEqual(
+        result.map((x) => x.date_count),
+        [1, 1, 2, 1]
+    );
+
+    result = await server.performRPC("", {
+        model: "bar",
+        method: "read_group",
+        args: [[]],
+        kwargs: {
+            fields: ["foo"],
+            domain: [],
+            groupby: ["date:week"],
+        },
+    });
+    assert.deepEqual(
+        result.map((x) => x["date:week"]),
+        ["W50 2016", "W43 2016", "W15 2016"]
```

Comment:
maybe add a record that has a different ISO year than normal year ?
e.g. 30 dec 2019 should be in W01 2020